MoneySaver.Api.Services/Implementation/TransactionService.cs

```diff
@@ -0,0 +1,72 @@
+﻿using AutoMapper;
+using MoneySaver.Api.Data;
+using MoneySaver.Api.Data.Repositories;
+using MoneySaver.Api.Services.Contracts;
+using MoneySaver.Api.Services.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace MoneySaver.Api.Services.Implementation
+{
+    public class TransactionService : ITransactionService
+    {
+        private IRepository<Transaction> transactionRepository;
+        private IMapper mapper;
+
+        public TransactionService(IRepository<Transaction> transactionRepository, IMapper mapper)
+        {
+            this.transactionRepository = transactionRepository;
+            this.mapper = mapper;
+        }
+
+        public TransactionModel CreateTransaction(TransactionModel transactionModel)
+        {
+
+            Transaction transaction = mapper.Map<Transaction>(transactionModel);
+            this.transactionRepository.AddAsync(transaction);
+
+            return transactionModel;
+        }
+
+        public List<TransactionModel> GetAllTransactions()
+        {
+            var transactions = this.transactionRepository.GetAll().ToList();
+            List<TransactionModel> transactionModels = new List<TransactionModel>();
+
+            for (int i = 0; i < transactions.Count; i++)
+            {
+                TransactionModel transactionModel = mapper.Map<TransactionModel>(transactions[i]);
+                transactionModels.Add(transactionModel);
+            }
```

Comment:
Instead of creating in memory one list with the transactions from the database and then another one with type "TransactionModel", you can use the LINQ. 

var transactionModels = this.transactionRepository.GetAll().Select(m => maper.Map<TransactionModel>(m))